Install page improvement based on feedback (#15487)
docker/docker.github.io: compose/install/index.md

keywords: compose, orchestration, install, installation, docker, documentation
title: Install Docker Compose
toc_max: 2
redirect_from:
- /compose/compose-desktop/
---

> On this page you can find information on how to get and install Compose.

Reviewer: ```suggestion
```

I think this page is short and self-evident enough not to require this explanation.

Author: Agree. Just keeping what we're doing as a standard in the docs. Usually it helps with the weirdness of having content start right after the page's title (from a visual perspective).

Author: See closing comment.

## Install Compose

> **To get Compose:**

Reviewer: ```suggestion
```

This seems redundant with the header, but doesn't add much additional info.

> * If you have Docker Desktop, you've got a full Docker installation, including Compose.

Reviewer: @dockertopia, is there a simple check the user can do to confirm they have docker compose? They should just be able to run `docker compose` right?

> * If you have Docker Desktop, you've got a full Docker installation, including Compose.

Author: See closing comment.

> * If you don't have Docker yet installed, or just need to install Compose, check the options below:

Reviewer: ```suggestion
```

I think the "If you don't have Docker yet installed" should be included with the prior sentence about Docker Desktop. So that way, we address DD entirely with the first point:

> If you've previously installed Docker Desktop, then you already have a full Docker installation that includes Compose. (If not, download links for Docker Desktop by OS are in the table below.)

Then, that leaves the statement: 
> If you…just need to install Compose, check the options below:

This could be more helpful. Under what circumstances would the user need to install Compose outside Docker Desktop? And how do they tell the difference between whether they should install it as a plugin, or standalone?

(I address this point a bit in the "Table Description" feedback below)

Author: See closing comment.

{% assign yes = '![yes](/images/green-check.svg){: .inline style="height: 14px; margin: 0 auto; align=right"}' %}

| Platform       | Docker Desktop                 | Compose CLI Plugin      | Compose Standalone  |
|:---------------|:------------------------------:|:-----------------------:|:-------------------:|
|Linux (64b)     |{{ yes }} [Install](../../desktop/install/linux-install.md)|{{ yes }} [Install](./compose-linux.md)|{{ yes }} [Install](./compose-linux.md#install-compose-standalone)|
|Mac             |{{ yes }} [Install](../../desktop/install/mac-install.md)    | - | - |
|Windows         |{{ yes }} [Install](../../desktop/install/windows-install.md)| - | - |
|Windows Server  | - | - |{{ yes }} [Install](#install-compose-on-windows-server)|

> **Table description:**
> * **Linux(64bit):** *option1)* Install [Docker Desktop for Linux](../../desktop/install/linux-install.md) or, *option2)* [Install Docker Engine and the CLI](../../engine/install/#server) and then [install Compose](compose-linux.md).
> * **macOS:** [Install Docker Desktop](../../desktop/install/mac-install/).
> * **Windows:** [Install Docker Desktop](../../desktop/install/windows-install/).
> * **Windows Server:** [Install Docker Engine and the CLI](../../engine/install/binaries/#install-server-and-client-binaries-on-windows) and then [install Compose](#install-compose-on-windows-server).
> 

Reviewer: This "table description" seems redundant with the information that's already in the table.  

Is this intended for accessibility, though? If so, that is unfortunately inconsistent with other instances of table usage. (e.g., on the DD [Linux Install](https://docs.docker.com/desktop/install/linux-install/#supported-platforms) page). 😞 

I would suggest that we remove from this section the repeated Docker Desktop download links, since it detracts from the simplicity and clarity of the links already in the table.

Without those links, it leaves the instructions for Linux and Windows Server. How about including them as clear sections?

### Note on Linux

```suggestion
### Note on Linux

On Linux(64bit), you may want to install Docker Engine separately from Compose, instead of the approach of installing everything with Docker Desktop. [This gives you more control over your install location and version of Compose.]

 1. [Install Docker Engine and the CLI](../../engine/install/#server) 
 2. [install Compose](compose-linux.md).

Alternately, you may want to [install Compose standalone], without the pre-requisite of installing the Docker Engine. 

###  Note on Windows Server
(This then replaces "Install Compose on Windows Server")
```

> [This gives you more control over your install location and version of Compose.]

I just put this here as a placeholder for trying to explain a common use case for bypassing the Docker Desktop install. 

Is there more we can say here to address "Under what circumstances would the user need to install Compose outside Docker Desktop? And how do they tell the difference between whether they should install it as a plugin, or standalone?"

Author: See closing comment.
